roles/collect/tasks/main.yml

```
@@ -194,6 +181,42 @@
     var: source_commit
   when: debug
 
+- name: Check if trusted-ca-bundle exists
+  k8s_info:
+    api_version: v1
+    kind: ConfigMap
+    namespace: "{{ namespace }}"
+    name: "trusted-ca-bundle"
+  register: trusted_ca_bundle_check
+
+- name: debug the trusted-ca-bundle
+  debug:
+    var: trusted_ca_bundle_check
+  when: debug
+
+- name: Create trusted-ca-bundle if it doesn't exist
+  k8s:
+    namespace: "{{ namespace }}"
+    state: present
+    src: '{{ ansible_env.HOME }}/roles/collect/files/trusted_ca_certmap.yaml'
+  when: not trusted_ca_bundle_check.resources
```

Do we need to check if it exists? Isn't that the point of `present` is that if it is there it doesn't do something and if it is not it will add it?
```suggestion
- name: Create trusted-ca-bundle if it doesn't exist
  k8s:
    namespace: "{{ namespace }}"
    state: present
    src: '{{ ansible_env.HOME }}/roles/collect/files/trusted_ca_certmap.yaml'
```